KarhooUISDK/Screens/CheckoutScreen/NewCheckout/Workers/KarhooNewCheckoutBookingWorker.swift

```diff
@@ -191,4 +334,16 @@ final class KarhooNewCheckoutBookingWorker: NewCheckoutBookingWorker {
                 //            self.bookingState = .failure(ErrorModel(message: "something_went_wrong", code: "0231"))
             )})
     }
+
+    private func isLoyaltyEnabled() -> Bool {
+        let loyaltyId = userService.getCurrentUser()?.paymentProvider?.loyaltyProgamme.id
+        return loyaltyId != nil && !loyaltyId!.isEmpty && LoyaltyFeatureFlags.loyaltyEnabled
+    }
+
+    private func isKarhooUser() -> Bool {
+        switch sdkConfiguration.authenticationMethod() {
+        case .karhooUser: return true
+        default: return false
+        }
+    }
```

Comment:
of simpler:
`private func isKarhooUser() -> Bool {
   sdkConfiguration.authenticationMethod() = .karhooUser
}`

Reply:
Unfortunately, the `AuthenticationMethod` enum is not `Equatable` so I can not compare instances.